src/chain_sync/tipset_syncer.rs

```diff
@@ -851,15 +851,20 @@ async fn sync_headers_in_reverse<DB: Blockstore + Sync + Send + 'static>(
             .await
             .map_err(TipsetRangeSyncerError::NetworkTipsetQueryFailed)?;
 
-        for tipset in network_tipsets {
-            // This could happen when the `until_epoch` is a null epoch
-            if tipset.epoch() < until_epoch {
-                break 'outer;
-            }
+        let callback = |tipset: Arc<Tipset>| {
             validate_tipset_against_cache(bad_block_cache, tipset.key(), &accepted_blocks)?;
             accepted_blocks.extend(tipset.cids());
             tracker.write().set_epoch(tipset.epoch());
             pending_tipsets.push(tipset);
+            Ok(())
+        };
+        // Breaks the loop when `until_epoch` is overreached, which happens
+        // when there are null tipsets in the queried range.
+        // Note that when the `until_epoch` is null, the outer while condition
+        // is always true, and it relies on this flag to break the loop
```

Comment:
Does this comment match the implemented logic?

Reply:
Yes. Which part do you think is in doubt?

Reply:
What is the `flag`?

Reply:
Oh, it refers to the returned boolean value. How about 
`and it relies on the returned boolean flag(until epoch is overreached) to break the loop`

Reply:
fixed.

Reply:
I wouldn't use the word `flag`; it suggests (subjectively) that there is a mutable, boolean variable flying around upon which the outer loop relies. `return value` would be more fitting here.

Reply:
Fixed